Tests/AK/TestFixedArray.cpp

```diff
@@ -23,7 +25,59 @@ TEST_CASE(ints)
     EXPECT_EQ(ints[0], 0);
     EXPECT_EQ(ints[1], 1);
     EXPECT_EQ(ints[2], 2);
+}
+
+TEST_CASE(swap)
+{
+    FixedArray<int> first = FixedArray<int>::must_create_but_fixme_should_propagate_errors(4);
+    FixedArray<int> second = FixedArray<int>::must_create_but_fixme_should_propagate_errors(5);
+    first[3] = 1;
+    second[3] = 2;
+    first.swap(second);
+    EXPECT_EQ(first.size(), 5u);
+    EXPECT_EQ(second.size(), 4u);
+    EXPECT_EQ(first[3], 2);
+    EXPECT_EQ(second[3], 1);
+}
+
+TEST_CASE(move)
+{
+    FixedArray<int> moved_from_array = FixedArray<int>::must_create_but_fixme_should_propagate_errors(6);
+    FixedArray<int> moved_to_array(move(moved_from_array));
+    EXPECT_EQ(moved_to_array.size(), 6u);
+    EXPECT_CRASH("Index into a moved-from FixedArray", [&] {
+        moved_from_array[0] = 0xbad;
+        return Test::Crash::Failure::DidNotCrash;
+    });
+}
+
+TEST_CASE(no_allocation)
+{
+    FixedArray<int> array = FixedArray<int>::must_create_but_fixme_should_propagate_errors(5);
+    EXPECT_NO_CRASH("Assigments", [&] {
+        NoAllocationGuard guard;
+        array[0] = 0;
+        array[1] = 1;
+        array[2] = 2;
+        array[4] = array[1];
+        array[3] = array[0] + array[2];
+        return Test::Crash::Failure::DidNotCrash;
+    });
+
+    EXPECT_NO_CRASH("Move", [&] {
+        FixedArray<int> moved_from_array = FixedArray<int>::must_create_but_fixme_should_propagate_errors(6);
+        NoAllocationGuard guard;
+        FixedArray<int> moved_to_array(move(moved_from_array));
+        // We need to ensure that this destructor runs before the FixedArray destructor.
+        guard.~NoAllocationGuard();
+        return Test::Crash::Failure::DidNotCrash;
+    });
 
-    ints.clear();
-    EXPECT_EQ(ints.size(), 0u);
+    EXPECT_NO_CRASH("Swap", [&] {
+        FixedArray<int> target_for_swapping;
+        NoAllocationGuard guard;
+        array.swap(target_for_swapping);
+        guard.~NoAllocationGuard();
```

Comment:
Just put the code in an explicit scope, much nicer than manually calling destructors. 
```suggestion

		{
            NoAllocationGuard guard;
            array.swap(target_for_swapping);
        }
```